### PR TITLE
1856 CGR operates immediately* if none of the forming corporations operated

### PR DESCRIPTION
**Pull request by ventusignis**
This also fixes a few bugs I found while testing

### CODE REVIEW

File: lib/engine/game/g_1856/game.rb [hunk 1]
```diff
@@ -1600,13 +1600,32 @@ def national_token_swap
           # Other tokens second, ignoring duplicates from the home token set
           @nationalized_corps.each do |corp|
             corp.tokens.each do |token|
-              next if !token.used || !token.city || home_bases.include?(token.city.hex)
+              next if !token.used || !token.city || home_bases.map(&:hex).include?(token.city.hex)
```
Comment (tobymao) [inline lib/engine/game/g_1856/game.rb:1603]: on here and 1638, probably more efficient to do

home_bases.any? { |base| base.hex == token.city.hex }

File: lib/engine/game/g_1856/game.rb [hunk 2]
```diff
@@ -1649,6 +1668,12 @@ def post_corp_nationalization
             c.close!
           end
 
+          earliest_index = @nationalized_corps.map { |n| @round.entities.find_index(n) }.min
+          current_corp_index = @round.entities.find_index(train_by_id('6-0').owner)
```
Comment (tobymao) [inline lib/engine/game/g_1856/game.rb:1672]: shorter method, index

File: lib/engine/game/g_1856/step/remove_tokens.rb [hunk 3]
```diff
@@ -26,29 +26,37 @@ def active_entities
...
           def corps
-            [pending_removal[:corp]]
+            [(duplicate_token && duplicate_token[:corp]) || pending_removal[:corp]]
           end
 
           def can_replace_token?(entity, token)
-            available_hex(entity, token.city.hex) && corps.include?(token.corporation)
+            available_hex(entity, token.city.hex) && corps.include?(token.corporation) &&
+            # duplicate tokens first
+            (duplicate_token[:tokens].nil? || duplicate_token[:tokens].include?(token))
```
Comment (tobymao) [inline lib/engine/game/g_1856/step/remove_tokens.rb:59]: !duplicate_tokens[:tokens]

File: lib/engine/game/g_1856/game.rb (hunk 1 above)
Comment (ventusignis) [reply lib/engine/game/g_1856/game.rb]: Changed

File: lib/engine/game/g_1856/game.rb (hunk 2 above)
Comment (ventusignis) [reply lib/engine/game/g_1856/game.rb]: Changed

File: lib/engine/game/g_1856/step/remove_tokens.rb (hunk 3 above)
Comment (ventusignis) [reply lib/engine/game/g_1856/step/remove_tokens.rb]: fixed

**Pull request merged**